### PR TITLE
update to v0.5.1, fix deps

### PR DESCRIPTION
**Pull request by ngam**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,13 +20,14 @@ requirements:
...
     - python >=3.6
-    - dataclasses
-    - jax >=0.2.6
+    - jax >=0.3.2
     - matplotlib-base
     - msgpack-python
     - numpy >=1.12
-    - optax >=0.0.6
+    - optax
+    - rich ~=11.1.0
```
Comment (PhilipVinc) [inline recipe/meta.yaml:28]:
```suggestion
    - rich >=11.1.0
```
Since conda does not work with upper bounds, let's not have them anywhere...

**Review by PhilipVinc — APPROVED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -20,13 +20,14 @@ requirements:
...
   run:
     - python >=3.6
-    - dataclasses
-    - jax >=0.2.6
+    - jax >=0.3.2
     - matplotlib-base
     - msgpack-python
     - numpy >=1.12
-    - optax >=0.0.6
+    - optax
```
Comment (PhilipVinc) [inline recipe/meta.yaml:27]:
```suggestion
    - optax >=0.0.6
```
Why did you remove this?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -20,13 +20,14 @@ requirements:
...
+    - rich >=11.1.0
```
Comment (PhilipVinc) [inline recipe/meta.yaml:28]:
```suggestion
    - rich ~=11.1.0
```

**Comment by ngam** [comment]
@PhilipVinc thank you so much for merging this (and the ones in chex and optax). I should have explained the stuff better, but I just followed the latest from upstream. I noticed that there was an issue of solver resolution and I realized some of these deps need updating --- funny how such little deps can cause a jam actually, but oh well...

Let me know if you'd like any help maintaining around here. We have recently add a cuda-supported jaxlib on conda-forge fyi 🥳

```
CONDA_OVERRIDE_CUDA="11.2" conda create -n cudafuljaxlib jaxlib==*=*cuda* jax flax
```

gets all you the fun :)
